synapse/http/client.py

```diff
@@ -937,12 +937,12 @@ class InsecureInterceptableContextFactory(ssl.ContextFactory):
     Do not use this since it allows an attacker to intercept your communications.
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
         self._context = SSL.Context(SSL.SSLv23_METHOD)
         self._context.set_verify(VERIFY_NONE, lambda *_: False)
 
     def getContext(self, hostname=None, port=None):
         return self._context
 
-    def creatorForNetloc(self, hostname, port):
+    def creatorForNetloc(self, hostname: bytes, port: int):
         return self
```

Comment:
This is supposed to return `IOpenSSLClientConnectionCreator`, but as far as I can see `InsecureInterceptableContextFactory` doesn't implement that interface. I chose to ignore this because this is meant to be for testing only.

Reply:
I think I ran into this too, there's a bit of funkiness in some of these interfaces that they don't all seem to be accurate.